kinto/__main__.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+import sys
+import subprocess
+
+from pyramid.paster import bootstrap
+
+
+def kinto_migrate(env):
+        registry = env['registry']
+        for backend in ('cache', 'storage', 'permission'):
+                if hasattr(registry, backend):
+                        getattr(registry, backend).initialize_schema()
+
+
+def main(args=None):
+        """The main routine."""
+        if args is None:
+                args = sys.argv[1:]
+
+        parser = argparse.ArgumentParser(description="Kinto commands")
+        subparsers = parser.add_subparsers(title='subcommands',
+                                           description='valid subcommands',
+                                           help='start/migrate')
+
+        parser_init = subparsers.add_parser('init')
+        parser_init.set_defaults(which='init')
+
+        parser_migrate = subparsers.add_parser('migrate')
+        parser_migrate.set_defaults(which='migrate')
+
+        parser_start = subparsers.add_parser('start')
+        parser_start.set_defaults(which='start')
+
+        args = vars(parser.parse_args())
+
+        if args['which'] == 'init':
+                """Still needs to setup the configuration options"""
+                pass
+        elif args['which'] == 'migrate':
+                env = bootstrap('config/kinto.ini')
+                kinto_migrate(env)
+                print("running migrations")
+        elif args['which'] == 'start':
+                subprocess.call(["python", "-m", "pyramid.scripts.pserve",
+                                "config/kinto.ini", "--reload"])
```

Comment:
Small remark : I believe you could look into this `pyramid/scripts/pserve.py` and see if can call a function instead of running a subcommand.